### DISCUSSION TITLE
Breathwork or Mediation Activities not Downloading

### DISCUSSION THREAD

**Discussion by runthesix** [philosowaffle/peloton-to-garmin#216] [Q&A] [answered]
Absolutely LOVIng P2G so far! I noticed all of my activities download except meditation. These can be mapped to Garmin breathwork activities. Is there a reason these specific activities don't download?

**Comment by Vicore1** [comment, marked as answer]
So by default if you are using the template config (under the Peloton section) provided they are excluded by default. If you remove this exclusion they map over as yoga. As for why they are not/can not be mapped to something else that is a question for smarter folks. I will say even after they are moved over as yoga they cannot be switched in connect to breathwork. It's not an option in the UI.  You could also open a feature request for that functionality to be added, but I'm guessing the reason it's not already there is because it can't be done, easily.

**Reply by philosowaffle** [reply]
The exercise types we have to work with can be found [here](https://github.com/philosowaffle/peloton-to-garmin/blob/master/src/Garmin/FitSdk/Dynastream/Fit/Profile/Types/SubSport.cs#L25). This code is the library published directly by Garmin and unfortunately, breathwork is not an option in this list.

Now, the version of the library P2G is using is about 6months old, so there is a chance a new version of the library includes Breathwork as an option.  I've added this as something to check for when I update to the latest version of the library, you can follow that work [here](https://github.com/philosowaffle/peloton-to-garmin/issues/213).

**Reply by runthesix** [reply]
Hi @Vicore1 and @philosowaffle. Thank you so much for the feedback! Exited to see where this goes? I was perfectly happy creating the activities manually, but this would be a really great update. =)

**Comment by danielsfrazier** [comment]
It appears that the latest FitSDKRelease_21.67.00 SDK does have breathing
exercises.

Breathing = 62,

**Comment by runthesix** [comment]
Hey @philosowaffle, thanks for updating the SubSport.cs file to include breathing. What's the best way to update my clone? Do I just overwrite the file or do I clone it again and rebuild? Thanks!

**Reply by philosowaffle** [reply]
Assuming when you cloned the repository, you did so using the command line and something like:
```
 > git clone (repo url)
```

Then all you need to do to get the latest version is navigate to inside your `peloton-to-garmin` folder. This is where the code is cloned locally.  Then just run:
```
> git fetch
> git pull
```

The first command basically asks Github if there are any new changes out there. And the second command pulls those changes down locally, updating all of the appropriate files for you automagically.

Then you're good to rebuild and run the tool!

P.S If you copied the code down a different way other than `git clone`, just let me know what steps you followed and I can give you appropriate steps to update.

**Reply by runthesix** [reply]
@philosowaffle, Ahh, actually cloning it would have made more sense. I actually downloaded the code and then followed the rest of the steps here: https://philosowaffle.github.io/peloton-to-garmin/install/source.html

I'm happy to start over by properly cloning it via git clone if that makes it easier?

**Reply by philosowaffle** [reply]
There's no harm in downloading the source code manually, entirely up to you and what you're most comfortable with.

But the steps to update for the manual download would be:

1. Copy your `configuration.local.json` file somewhere safe
2. Delete the `peloton-to-garmin` folder
3. Download a fresh copy of the repo
4. Do steps 4 & 6 in the [build from source](https://philosowaffle.github.io/peloton-to-garmin/install/source.html) instructions

**Reply by runthesix** [reply]
THANK YOU! I blew everything out and started fresh by cloning it the "right way." Breathwork activities are downloading now! 

Really appreciate this. Thanks again!

![Screen Shot 2022-02-05 at 8 59 40 PM](https://user-images.githubusercontent.com/97648335/152665186-377be508-dd17-42eb-8abd-70f8ab78d2f9.png)
